aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -765,49 +763,38 @@ _scaled_dot_product_flash_attention_cpu(
     "scaled_dot_product_attention_flash_attention: Currently do not support dropout > 0");
   TORCH_CHECK((query.size(3) == value.size(3)) && (key.size(3) == value.size(3)),
     "scaled_dot_product_attention_flash_attention: Q/K/V should have the same head size");
-  TORCH_CHECK(return_debug_mask == false,
-    "scaled_dot_product_attention_flash_attention: Currently do not support 'return_debug_mask'");
+  TORCH_CHECK(!attn_mask.has_value() ||
+          dtype == attn_mask.value().scalar_type(),
+    "scaled_dot_product_attention_flash_attention: Attention mask is the same data type as query");
+  TORCH_CHECK(!attn_mask.has_value() ||
+          (attn_mask.value().dim() >= 2 && attn_mask.value().dim() <= 4),
+    "scaled_dot_product_attention_flash_attention: Attention mask dim in {2, 3, 4}");
```

Comment:
I think for consistency with the math and the cuda kernels we should not do the 3d case

Reply:
Sure. Thanks and modified.